test/fakes/fakehealthchecker/checker.go

```diff
@@ -0,0 +1,63 @@
+package fakehealthchecker
+
+import (
+	"fmt"
+
+	"github.com/spiffe/spire/pkg/common/health"
+)
+
+type Checker struct {
+	checkables map[string]health.Checkable
+}
+
+var _ health.Checker = (*Checker)(nil)
+
+func New() *Checker {
+	return &Checker{
+		checkables: make(map[string]health.Checkable),
+	}
+}
+
+func (c *Checker) AddCheck(name string, checkable health.Checkable) error {
+	if _, ok := c.checkables[name]; ok {
+		return fmt.Errorf("check %q has already been added", name)
+	}
+	c.checkables[name] = checkable
+	return nil
+}
+
+func (c *Checker) LiveState() (bool, interface{}) {
```

Comment:
I propose we either let unit-tests assert their health behavior through these methods or the state returned by RunChecks(), but not both.

If we like these methods better, then I further propose that we try and share some code with the `health` package so that the live and ready state and details are calculated in the same way. As it is written now, there is no guarantee that these functions behave in the same way as those in the `health` package, so there is not as strong a guarantee by a unit-test that the health check for that unit will ultimately result in the correct live and ready state and details.

Reply:
For now, I think we will use RunChecks() for the unit tests. I'll rewrite these functions when I'll write health subsystem unit tests.